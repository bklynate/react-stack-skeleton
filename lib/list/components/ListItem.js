"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ListItem;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * An item in the todo list.
 */
function ListItem(props) {
  var label = props.label,
      value = props.value,
      completed = props.completed,
      onChangeValue = props.onChangeValue,
      onSwitchStatus = props.onSwitchStatus,
      otherProps = _objectWithoutProperties(props, ["label", "value", "completed", "onChangeValue", "onSwitchStatus"]);

  return _react2.default.createElement(
    "li",
    { className: completed ? "completed" : undefined },
    _react2.default.createElement(
      "div",
      { className: "view" },
      _react2.default.createElement("input", { className: "toggle", type: "checkbox", checked: completed, onChange: onSwitchStatus }),
      _react2.default.createElement(
        "label",
        null,
        label
      ),
      _react2.default.createElement("button", { className: "destroy" })
    ),
    _react2.default.createElement("input", { className: "edit", value: value, onChange: onChangeValue })
  );
}