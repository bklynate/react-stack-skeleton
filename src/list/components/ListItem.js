/* @flow */
import React from 'react';

type Props = {
  value: string,
  completed?: bool,
  editing?: bool,
  onChangeValue: (e: SyntheticInputEvent) => void,
  onClickSwitchStatus: (e: SyntheticInputEvent) => void,
  onClickDelete: (e: SyntheticEvent) => void,
};

/**
 * An item in the todo list.
 */
export default function ListItem(props: Props): React.Element<*> {
  const { value, completed, editing, onChangeValue, onClickSwitchStatus, onClickDelete, ...otherProps } = props;

  return (
    <li className={[completed? 'completed': null, editing? 'editing': null ].filter((str) => !!str).join(' ')}>
      <div className="view">
        <input className="toggle" type="checkbox" checked={completed} onChange={onClickSwitchStatus} />
        <label>{value}</label>
        <button className="destroy" onClick={onClickDelete}></button>
      </div>
      <input className="edit" value={value} onChange={onChangeValue} />
    </li>
  );
}
