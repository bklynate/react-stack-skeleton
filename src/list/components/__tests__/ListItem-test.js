import React from 'react';
import renderer from 'react-test-renderer';

import ListItem from '../ListItem';

test('it displays the given value in the label and input for the item.', () => {
  const component = renderer.create(
    <ListItem value="Test Value" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it shows the input as un-checked if the item is not completed.', () => {
  const component = renderer.create(
    <ListItem />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it shows the input as checked if the item is completed.', () => {
  const component = renderer.create(
    <ListItem completed />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it shows the items in editing mode if the item has editing prop set.', () => {
  const component = renderer.create(
    <ListItem editing />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it calls the onChangeValue function if the item is changed.', () => {
  const onChangeFn = jest.fn();
  const component = renderer.create(
    <ListItem onChangeValue={onChangeFn} />
  );
  const tree = component.toJSON();
  const itemValueInput = tree.children[1];

  itemValueInput.props.onChange();
  expect(onChangeFn).toHaveBeenCalledTimes(1);
});

test('it calls the onSwitchStatus function if the item\'s checksbox is clicked.', () => {
  const onSwitchFn = jest.fn();
  const component = renderer.create(
    <ListItem onClickSwitchStatus={onSwitchFn} />
  );
  const tree = component.toJSON();
  const itemCheckboxInput = tree.children[0].children[0];

  itemCheckboxInput.props.onChange();
  expect(onSwitchFn).toHaveBeenCalledTimes(1);
});

test('it calls the onClickDelete function if the item is changed.', () => {
  const onClickFn = jest.fn();
  const component = renderer.create(
    <ListItem onClickDelete={onClickFn} />
  );
  const tree = component.toJSON();
  const distoryButton = tree.children[0].children[2];

  distoryButton.props.onClick();
  expect(onClickFn).toHaveBeenCalledTimes(1);
});
