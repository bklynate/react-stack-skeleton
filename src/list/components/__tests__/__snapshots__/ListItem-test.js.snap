exports[`test it displays the given value in the label and input for the item. 1`] = `
<li
  className="">
  <div
    className="view">
    <input
      checked={undefined}
      className="toggle"
      onChange={undefined}
      type="checkbox" />
    <label>
      Test Value
    </label>
    <button
      className="destroy"
      onClick={undefined} />
  </div>
  <input
    className="edit"
    onChange={undefined}
    value="Test Value" />
</li>
`;

exports[`test it shows the input as checked if the item is completed. 1`] = `
<li
  className="completed">
  <div
    className="view">
    <input
      checked={true}
      className="toggle"
      onChange={undefined}
      type="checkbox" />
    <label />
    <button
      className="destroy"
      onClick={undefined} />
  </div>
  <input
    className="edit"
    onChange={undefined}
    value={undefined} />
</li>
`;

exports[`test it shows the input as un-checked if the item is not completed. 1`] = `
<li
  className="">
  <div
    className="view">
    <input
      checked={undefined}
      className="toggle"
      onChange={undefined}
      type="checkbox" />
    <label />
    <button
      className="destroy"
      onClick={undefined} />
  </div>
  <input
    className="edit"
    onChange={undefined}
    value={undefined} />
</li>
`;

exports[`test it shows the items in editing mode if the item has editing prop set. 1`] = `
<li
  className="editing">
  <div
    className="view">
    <input
      checked={undefined}
      className="toggle"
      onChange={undefined}
      type="checkbox" />
    <label />
    <button
      className="destroy"
      onClick={undefined} />
  </div>
  <input
    className="edit"
    onChange={undefined}
    value={undefined} />
</li>
`;
